101-machine-learning-compute-create-computeinstance/azuredeploy.json

```diff
@@ -0,0 +1,106 @@
+            "defaultValue": "",
+            "type": "securestring",
+            "metadata": {
+              "description": "The admin username needed to access the virtual machine."
+            }
+        },
+        "adminPublicKey": {
+            "defaultValue": "",
+            "type": "securestring",
+            "metadata": {
+              "description": "The SSH public key of the administrator user account."
+            }
+        },
+        "sshPublicAccess": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+              "description": "Indicates the state of the public SSH port."
+            }
+        },
+        "sshPort": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+              "description": "The exposed port for the compute instance."
+            }
+        },
+        "vmSize": {
+            "defaultValue": "Standard_DS3_v2",
+            "type": "string",
+            "metadata": {
+              "description": "The size of agent VMs. More details can be found here: https://aka.ms/azureml-vm-details."
+            }
+        },
+        "vnetResourceGroupName": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the resource group which holds the VNET to which you want to inject your compute in."
+            }
+        },
+        "vnetName": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the vnet which you want to inject your compute in."
+            }
+        },
+        "subnetName": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the subnet inside the VNET which you want to inject your compute in."
+            }
+        }
+    },
+    "variables": {
+        "sshSettings": {
+            "sshPublicAccess": "[parameters('sshPublicAccess')]",
+            "adminUserName": "[parameters('adminUserName')]",
+            "sshPort": "[parameters('sshPort')]",
+            "adminPublicKey": "[parameters('adminPublicKey')]"
+        }
+    },
+    "resources": [        
+        {
+            "type": "Microsoft.MachineLearningServices/workspaces/computes",
+            "name": "[concat(parameters('workspaceName'), '/', parameters('computeName'))]",
+            "apiVersion": "2020-04-01",
+            "location" : "[parameters('location')]",
+            "properties": {
+                "computeType": "ComputeInstance",
+                "properties":
+                {
+                    "vmSize" : "[parameters('vmSize')]",
+                    "subnet" : "[if(not(empty(parameters('subnetName'))), [resourceId(parameters('vnetResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('subnetName'))], json('null'))]",
```

Comment:
fix this condition as in the other sample: https://github.com/Azure/azure-quickstart-templates/blob/master/101-machine-learning-compute-create-amlcompute/azuredeploy.json#L105

Reply:
fixed